channels/rail/client/rail_orders.c

```diff
@@ -650,7 +650,9 @@ UINT rail_order_recv(railPlugin* rail, wStream* s)
 		case RDP_RAIL_ORDER_EXEC_RESULT:
 			{
 				RAIL_EXEC_RESULT_ORDER execResult;
-				return rail_recv_exec_result_order(rail, &execResult, s);
+				error = rail_recv_exec_result_order(rail, &execResult, s);
+				free(execResult.exeOrFile.string);
```

Comment:
Oh, nevermind. I missed the `ZeroMemory` call in rail_recv_exec_result_order(). It is too sneaky placed near the variable declarations.

So there are no errors here, the free() call is fine.